backendprocesses.cpp

```diff
@@ -100,8 +100,25 @@ void BackendProcesses::threadProcedure()
     all_bytes_in_minute.push_back(bytes);
     all_bytes_in_minute.push_back("</bsr>");
 
+    // determine base path
+    QString basePath;
+    QFileInfo check_file_p1("../sc1-driver-io/file_sync_output/placeholder.txt");
+    QFileInfo check_file_p2("../solar-car-dashboard/file_sync_output/placeholder.txt");
+    if (check_file_p1.exists() && check_file_p1.isFile()) {
+        basePath = "../sc1-driver-io/file_sync_output/";
+    } else if (check_file_p2.exists() && check_file_p2.isFile()) {
+        basePath = "../solar-car-dashboard/file_sync_output/";
+    } else {
+        qDebug()<<"\n";
+        qDebug()<<"FATAL ERROR: unable to find correct file sync output directory";
+        qDebug()<<"   * Your build output is probably in a nonstandard directory";
+        qDebug()<<"\n";
+
+        QGuiApplication::quit();
```

Comment:
Don't need to quit the application without file sync.

Reply:
I agree that we shouldn't quit the application if file sync isn't available (which is why it's just a warning if the program can't find main.py). This particular check is to make sure that the designated folder for putting the output files exists before trying to write to that path (to avoid clobbering the filesystem). Would it be better if it disabled doing the write call when this path doesn't exist? (maybe still with a one-time warning in the constructor?)

If that's the solution we go with, it might also be good to add an extra sanity check (an additional if statement and error message/panic) in [this code](https://github.com/badgerloop-software/sc1-driver-io/blob/a6ff64105ff71ea0f8ff8b6d3473ebe5363413d8/DataProcessor/DataUnpacker.cpp#L41) since it's fundamentally the same issue as here